Tools/Scripts/build-and-analyze

```diff
@@ -0,0 +1,195 @@
+# WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS BE LIABLE FOR
+# ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+# OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+import argparse
+import os
+import subprocess
+import sys
+
+
+def args_for_additional_checkers(checker_names):
+    if not len(checker_names):
+        return []
+
+    return ['-analyzer-checker', ','.join(checker_names)]
+
+
+def make_analyzer_flags(output_dir, args):
+    additional_checkers = []
+    analyzer_flags = []
+
+    if args.disable_default_checkers:
+        # FIXME: Disable clang-tidy bugprone-* checkers with -analyzer-tidy-checker=-*
+        #   when that works.
+        # See: <https://clang.llvm.org/docs/analyzer/checkers.html>
+        all_checker_categories = [
+            'alpha',
+            'apiModeling',
+            # 'bugprone-infinite-loop',
+            # 'bugprone-move-forwarding-reference',
+            'core',
+            'cplusplus',
+            'deadcode',
+            'debug',
+            'fuchsia',
+            'nullability',
+            'optin',
+            'osx',
+            'security',
+            'unix',
+            'webkit',
+        ]
+        analyzer_flags.extend([
+            '-analyzer-disable-checker',
+            ','.join(all_checker_categories),
+        ])
+    else:
+        additional_checkers.extend([
+            'optin.cplusplus.UninitializedObject',
+            'optin.cplusplus.VirtualCall',
+        ])
+
+    if args.enable_webkit_checkers:
+        additional_checkers.extend([
+            'webkit.NoUncountedMemberChecker',
+            'webkit.RefCntblBaseVirtualDtor',
+            'webkit.UncountedLambdaCapturesChecker',
+            'alpha.webkit.UncountedCallArgsChecker',
+            'alpha.webkit.UncountedLocalVarsChecker',
+        ])
+
+    if args.only_smart_pointers:
+        additional_checkers.extend([
+            'alpha.webkit.UncountedCallArgsChecker',
+            'alpha.webkit.UncountedLocalVarsChecker',
+        ])
+
+    if additional_checkers:
+        analyzer_flags.extend(args_for_additional_checkers(additional_checkers))
+
+    if args.analyzer_budget_max_nodes is not None:
+        analyzer_flags.extend([
+            '-analyzer-config',
+            'max-nodes={}'.format(args.analyzer_budget_max_nodes),
+        ])
+
+    return ' '.join(analyzer_flags)
+
+
+def scan_build_path(sdkroot):
+    """
+    Returns path to scan-build from the macosx.internal SDK.
+    The internal SDK must be used due to <rdar://53628544>.
```

Comment:
Change this comment to something like:

```suggestion
    Try to find scan-build in the SDK, else assume scan-build is in the
    user's PATH.  It may be downloaded from the clang/llvm repository.
```